File: php/features.php
<?php
declare(strict_types=1);

// TranslateFunnyLanguages SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/RatelimitFeature.php';
require_once __DIR__ . '/feature/RetryFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';
require_once __DIR__ . '/feature/TimeoutFeature.php';


class TranslateFunnyLanguagesFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new TranslateFunnyLanguagesBaseFeature();
            case "ratelimit":
                return new TranslateFunnyLanguagesRatelimitFeature();
            case "retry":
                return new TranslateFunnyLanguagesRetryFeature();
            case "test":
                return new TranslateFunnyLanguagesTestFeature();
            case "timeout":
                return new TranslateFunnyLanguagesTimeoutFeature();
            default:
                return new TranslateFunnyLanguagesBaseFeature();
        }
    }

    /**
     * Does a generated feature class back this name? False for a name only
     * an options extend instance can supply (the station adopt path) - the
     * constructor uses this to skip make_feature for such names instead of
     * adding a stray BaseFeature.
     */
    public static function has_feature(string $name): bool
    {
        switch ($name) {
            case "base":
            case "ratelimit":
            case "retry":
            case "test":
            case "timeout":
                return true;
            default:
                return false;
        }
    }
}

File: py/translatefunnylanguages_sdk/features.py
# TranslateFunnyLanguages SDK feature factory

from translatefunnylanguages_sdk.feature.base_feature import TranslateFunnyLanguagesBaseFeature
from translatefunnylanguages_sdk.feature.ratelimit_feature import TranslateFunnyLanguagesRatelimitFeature
from translatefunnylanguages_sdk.feature.retry_feature import TranslateFunnyLanguagesRetryFeature
from translatefunnylanguages_sdk.feature.test_feature import TranslateFunnyLanguagesTestFeature
from translatefunnylanguages_sdk.feature.timeout_feature import TranslateFunnyLanguagesTimeoutFeature


_FEATURES = {
    "base": lambda: TranslateFunnyLanguagesBaseFeature(),
    "ratelimit": lambda: TranslateFunnyLanguagesRatelimitFeature(),
    "retry": lambda: TranslateFunnyLanguagesRetryFeature(),
    "test": lambda: TranslateFunnyLanguagesTestFeature(),
    "timeout": lambda: TranslateFunnyLanguagesTimeoutFeature(),
}


def _make_feature(name):
    factory = _FEATURES.get(name)
    if factory is not None:
        return factory()
    return _FEATURES["base"]()


# True when this SDK was generated with the named feature class - the
# constructor's tolerance for extend-carried features reads this (an
# active name with no generated class must not become a BaseFeature
# stray when an extend instance carries it).
def _has_feature(name):
    return name in _FEATURES
